File: backend/src/FinanceFlow.Api/Models/Domain/Transaction.cs
namespace FinanceFlow.Api.Models.Domain;

public class Transaction
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public DateTime Date { get; set; }
    public decimal Amount { get; set; }
    public string Details { get; set; } = string.Empty;
    public string AccountId { get; set; } = string.Empty;
    public string BudgetType { get; set; } = string.Empty;
    public string? BudgetPositionId { get; set; }
    public string UserId { get; set; } = string.Empty;

    public Account? Account { get; set; }
    public BudgetCategory? BudgetPosition { get; set; }
}

File: frontend/src/shared/types/transaction.ts
import type { BudgetType } from './common';

export interface Transaction {
  id: string;
  date: string; // ISO date
  amount: number;
  details: string;
  accountId: string;
  budgetType: BudgetType | '';
  budgetPositionId: string; // links to BudgetCategory.id
}
